src/Compilers/CSharp/Portable/Binder/Binder.WithQueryLambdaParametersBinder.cs

```diff
@@ -43,14 +43,14 @@ protected override BoundExpression BindRangeVariable(SimpleNameSyntax node, Rang
                         // the range variable maps directly to a use of the parameter of that name
                         var value = base.parameterMap[qv.Name];
                         Debug.Assert(value.Count == 1);
-                        translation = new BoundParameter(node, value.Single()) { WasCompilerGenerated = true };
+                        translation = new BoundParameter(node, value.Single(), associatedRangeVariable: qv) { WasCompilerGenerated = true };
```

Comment:
>{ WasCompilerGenerated = true } [](start = 108, length = 31)

I think you initial change too remove ```WasCompilerGenerated = true``` was the right thing to do. We simply need to override VisitRangeVariable in Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.NodeMapBuilder in such a way so that BoundRangeVariable.Value isn't visited.